cmd/p2p/crawl/crawl.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright © 2022 Polygon <engineering@polygon.technology>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package crawl
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"github.com/maticnetwork/polygon-cli/p2p"
+)
+
+type (
+	crawlParams struct {
+		Bootnodes string
+		Timeout   string
+		Threads   int
+		NetworkID int
+		NodesFile string
+		Database  string
+	}
+)
+
+var (
+	inputCrawlParams crawlParams
+)
+
+// crawlCmd represents the crawl command
+var CrawlCmd = &cobra.Command{
+	Use:   "crawl [nodes file]",
+	Short: "Crawl a network",
+	Long:  `This is a basic function to crawl a network.`,
+	Args:  cobra.MinimumNArgs(1),
+	PreRun: func(cmd *cobra.Command, args []string) {
+		inputCrawlParams.NodesFile = args[0]
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		inputSet, err := p2p.LoadNodesJSON(inputCrawlParams.NodesFile)
+		if err != nil {
+			return err
+		}
+
+		var cfg discover.Config
+		cfg.PrivateKey, _ = crypto.GenerateKey()
+		bn, err := p2p.ParseBootnodes(inputCrawlParams.Bootnodes)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to parse bootnodes")
+			return err
+		}
+		cfg.Bootnodes = bn
+
+		db, err := enode.OpenDB(inputCrawlParams.Database)
+		if err != nil {
+			return err
```

Comment:
may want to standardize log.Error() as we did with earlier condition before return